### PR TITLE
Implement corrected matching of h264 fmtp lines

### PR DESCRIPTION
**Pull request by k0nserv**
This is based on
https://datatracker.ietf.org/doc/html/rfc6184#section-8.1 and
libWebRTC's implementation.

### CODE REVIEW

File: src/format.rs [hunk 1]
```diff
@@ -824,6 +855,332 @@ impl<'a> From<&'a str> for Codec {
...
+    Level2_2 = 22_u8,
+    Level3 = 30_u8,
+    Level3_1 = 31_u8,
+    Level3_2 = 32_u8,
+    Level4 = 40_u8,
+    Level4_1 = 41_u8,
+    Level4_2 = 42_u8,
+    Level5 = 50_u8,
+    Level5_1 = 51_u8,
+    Level5_2 = 52_u8,
```
Comment (algesten) [inline src/format.rs:1152]: Turn off cargo fmt here and align.

File: src/format.rs [hunk 2]
```diff
@@ -824,6 +855,332 @@ impl<'a> From<&'a str> for Codec {
...
+            x if (Level2_2 as u8) == x => Ok(Level2_2),
+            x if (Level3 as u8) == x => Ok(Level3),
+            x if (Level3_1 as u8) == x => Ok(Level3_1),
+            x if (Level3_2 as u8) == x => Ok(Level3_2),
+            x if (Level4 as u8) == x => Ok(Level4),
+            x if (Level4_1 as u8) == x => Ok(Level4_1),
+            x if (Level4_2 as u8) == x => Ok(Level4_2),
+            x if (Level5 as u8) == x => Ok(Level5),
+            x if (Level5_1 as u8) == x => Ok(Level5_1),
+            x if (Level5_2 as u8) == x => Ok(Level5_2),
```
Comment (algesten) [inline src/format.rs:1178]: Same here. No cargo fmt

File: src/format.rs [hunk 3]
```diff
@@ -853,3 +1210,109 @@ impl std::ops::DerefMut for CodecConfig {
         &mut self.params
     }
 }
+
+#[cfg(test)]
+mod test {
+    use super::{Codec, CodecSpec, FormatParams, H264LevelIdc, IOPPattern, PayloadParams};
```
Comment (algesten) [inline src/format.rs:1216]: The standard in str0m is `use super::*` for `mod test {}`

**Review by algesten — CHANGES_REQUESTED** [review]
Looks good!

1. Some small formatting things.
2. Move this to `src/packet/h264_profile.rs` and export/import into `format.rs` accordingly.

Thanks!

File: src/format.rs [hunk 4]
```diff
@@ -824,6 +855,332 @@ impl<'a> From<&'a str> for Codec {
...
+            x if (Self::X6E as u8) == x => Ok(Self::X6E),
+            x if (Self::X7A as u8) == x => Ok(Self::X7A),
+            x if (Self::XF4 as u8) == x => Ok(Self::XF4),
+            _ => Err(()),
+        }
+    }
+}
+
+#[derive(Copy, Clone)]
+struct IOPPattern {
```
[the remark this answers is not in the document]
Comment (algesten) [reply src/format.rs:1072]: Yeah, we don't have that yet. If we find a second use for it we will.

[the remark this answers is not in the document]
Comment (algesten) [reply src/format.rs:1072]: Renaming it sounds good.

[the remark this answers is not in the document]
Comment (k0nserv) [reply src/format.rs:1072]: We do have a util module

[the remark this answers is not in the document]
Comment (algesten) [reply src/format.rs:1072]: Ha! Lol. Sure put it there :D

**Comment by k0nserv** [comment]
> I rather not change h264.rs to h264/mod.rs since that makes an asymmetry with all the other codec modes. Hence I want h264_profile.rs, for now.

I feel like the module paths in Rust are more important than the file paths, but I'll change it

**Pull request merged**